### PR TITLE
wire through humanreadable print flags

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

~~Work in progress... Opening PR now to gather feedback as this is implemented.~~

Begin implementing pieces needed to retrieve humanreadable printers from a set of flags.
Proposal: https://docs.google.com/document/d/19ZZFVe9oD1KQmk5uExggRWtRl_hKGfYnBXvHZJlgEro/edit#heading=h.pnvbfi14v4zz

cc @soltysh @deads2k @pwittrock

### CODE REVIEW

File: pkg/printers/humanreadable_flags.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	// TODO: handle sorting. importing "kubectl.SortingPrinter"
+
+	return p, true, nil
+}
+
+// AddFlags receives a *cobra.Command reference and binds
+// flags related to human-readable printing to it
+func (f *HumanPrintFlags) AddFlags(c *cobra.Command) {
+	if f.NoHeaders != nil {
+		c.Flags().BoolVar(f.NoHeaders, "no-headers", *f.NoHeaders, "When using the default or custom-column output format, don't print headers (default print headers).")
```
Comment (juanvallejo) [inline pkg/printers/humanreadable_flags.go]: @soltysh @deads2k The `no-headers` flag is also bound by custom-columns PrintFlags here: https://github.com/kubernetes/kubernetes/pull/61501/files#diff-031dfa2edd2855fceafd54e790b79f1cR102

File: pkg/printers/humanreadable_flags.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+package printers
+
+import (
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+)
+
+// HumanPrintFlags provides default flags necessary for template printing.
```
Comment (soltysh) [inline pkg/printers/humanreadable_flags.go]: It's not quite template printing, more like a user friendly form.

File: pkg/printers/humanreadable_flags.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+// Given the following flag values, a printer can be requested that knows
+// how to handle printing based on these values.
+type HumanPrintFlags struct {
+	NoHeaders *bool
+
+	// TODO: this is deprecated - remove this field in next version
```
Comment (soltysh) [inline pkg/printers/humanreadable_flags.go]: We deprecated that in 1.10. I think we can safely remove it right away, @deads2k objections? Or it'll be better to separate this out to its own pr.

File: pkg/printers/humanreadable_flags.go (hunk 1 above)
Comment (soltysh) [reply pkg/printers/humanreadable_flags.go]: Yeah, I guess we need to figure out the best way to solve this problem.

File: pkg/printers/humanreadable_flags.go (hunk 3 above)
Comment (juanvallejo) [reply pkg/printers/humanreadable_flags.go]: Thanks, removed this field

File: pkg/printers/humanreadable_flags.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/printers/humanreadable_flags.go]: Went ahead and added a parameter that sets this on the constructor func for the PrintFlags struct

**Comment by juanvallejo** [comment]
@soltysh thanks for the feedback.

I moved `sorting_printer.go` to the printers package. This way, we can avoid an import cycle when we begin handling sorting when printing at the PrintFlags level.

In order to print multiple resource kinds while keeping state here, we could add a wrapper to `humanreadable.go`, iterate through the sorted list of given runtime.Object, and call humanreadable#PrintObject on each one, keeping track of when to print headers, if resource kinds should be printed, etc. WDYT?

**Comment by juanvallejo** [comment]
@soltysh this PR should be good to go. It has been updated to take into account instantiating this set of PrintFlags at the command level.

Tests added.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve no-issue

**Comment by juanvallejo** [comment]
@soltysh fixed tests, mind re-tagging?

**Comment by juanvallejo** [comment]
@soltysh updated humanreadable printflags tests as discussed. PTAL

**Review by soltysh — APPROVED** [review]
/lgtm
